File: LC26_Remove Duplicates from Sorted Array_Double Pointer.java
class Solution {
public int removeDuplicates(int[] nums) {  // 双指针法
    if (nums.length == 0) return 0;
    int i = 0;
    for (int j = 1; j < nums.length; j++) {
        if (nums[j] != nums[i]) {
            i++;
            nums[i] = nums[j];
        }
    }
    return i + 1;  // 统计个数并输出
    }
}

File: LC26_Remove Duplicates from Sorted Array_Double Pointer.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        for (int num : nums) {
            if (i < 1 || nums[i - 1] < num) nums[i++] = num;
        }
        return i;
    }
};


// reference https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array-ii/solution/ju-yi-fan-er-ni-zhen-de-zhen-de-zhi-de-x-eicz/
